File: Python/Min_number_of_flips.py
"""
Idea:
assuming 2 cases:
if final string is 01010, or 10101, checking how many changes required to achieve, return min value
"""


class Solution:
    def minFlips(self, S):
        n = len(S)
        count = 0
        for i in range(n):
            if i % 2 == int(S[i]):
                count += 1

        return min(count, n - count)


if __name__ == '__main__':
    t = int(input())
    for i in range(t):
        S = input()
        Obj = Solution()
        ans = Obj.minFlips(S)
        print(ans)

File: C++/Min_number_of_flips.cpp
/*
Idea:
assuming 2 cases:
if final string is 01010, or 10101, checking how many changes required to achieve, return min value
*/


#include<bits/stdc++.h>
using namespace std;

int minFlips(string s);
int32_t main()
{
    int t; cin >> t;
    while (t--)
    {
        string s; cin >> s;

        cout << minFlips(s) << endl;
    }
}



int minFlips(string S)
{

    int count = 0, n = S.size();
    for (int i = 0; i < n; ++i)
        if ((i % 2) != (S[i] - '0'))
            count++;

    return min(count, n - count);

}